src/ui/token.js

```diff
@@ -158,30 +158,20 @@ class Token extends React.Component {
   }
 
   render() {
-    if (this.props._inHexGrid) {
-      return (
-        <Hex
-          x={this.state.x}
-          y={this.state.y}
-          z={this.state.z}
-          style={this.props.style}
-          onClick={this.props.onClick}
-        >
-          {this.props.children}
-        </Hex>
-      );
-    }
-
-    if (this.props._inGrid) {
-      return (
-        <Square
-          x={this.state.x}
-          y={this.state.y}
-          style={this.props.style}
-          onClick={this.props.onClick}
-        >
-          {this.props.children}
-        </Square>
+    if (this.props.template) {
+      return React.createElement(
+        this.props.template,
+        {
+          x: this.state.x,
```

Comment:
Now that we're calculating `center`, we no longer need to pass `x, y and z` to the template class, correct?

Reply:
No - this breaks the animation code. The `onClick` handler passes the cube coordinates up the hierarchy, and in the case of the `Runner` (inside storybook) then sets the state, which then gets interpolated inside Token.